### DISCUSSION TITLE
app install from google play show white screen

### DISCUSSION THREAD

**Discussion by jaychia0418** [flet-dev/flet#4084] [Q&A]
### Duplicate Check

- [X] I have searched the [opened issues](https://github.com/flet-dev/flet/issues) and there are no duplicates

### Describe the bug

I build apk and aab with this command
flet build --project myapp--product myapp--org myorg--include-packages flet_lottie flet_video --build-number 1 --build-version 1.0.0  apk
flet build --project myapp--product myapp--org myorg--include-packages flet_lottie flet_video --build-number 1 --build-version 1.0.0  aab

can run properly on my device, but when I upload and install from google play store, it show me white screen, why?

no error when build apk and aab

### Code sample

<details open><summary>Code</summary>

```python
[Paste your code here]
```

</details>

### To reproduce

1. build apk and aab
2. install on my device - no problem
3. upload aab to google play
4. install from google play
5. show white screen

### Screenshots / Videos

<details open>
<summary>Captures</summary>

[Upload media here]

</details>

### Operating System

Windows

### Operating system details

windows 11

### Flet version

0.23.2

### Regression

No, it isn't

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by Muddassir-Farooq-official** [comment]
Share app link

**Comment by jaychia0418** [comment]
https://play.google.com/store/apps/details?id=jay.chia.JayCounter

**Comment by jaychia0418** [comment]
> Share app link

apk
https://drive.google.com/file/d/16nhaDkO_96t_BBLK4l1zMJh1Qi6lbBFf/view?usp=sharing

aab
https://drive.google.com/file/d/1e-_fBk-aNnlo_41bYrYq7dwPiWiB-Hgw/view?usp=sharing

google play link (google play closed test)
https://play.google.com/store/apps/details?id=jay.chia.JayCounter

**Comment by Muddassir-Farooq-official** [comment]
You are right there is an issue with aab and showing white screen I think this issue occurred because after flet 0.21 same thing occur in apk but flet team resolve it I think they forgot to sort this aab issue have you clear cookies butter cache? @ndonkoHenri and @FeodorFitsner kindly have a look

**Comment by jaychia0418** [comment]
what is cookies butter cache and how to clear
if I change my flet version to 0.21.2 will fix this issue?

**Comment by Muddassir-Farooq-official** [comment]
It is fixed in Flet 0.23.1.

Please cleanup Cookiecutter cache before running flet build command.

On Windows:

rd /s /q %USERPROFILE%\.cookiecutters
On Mac/Linux:

rm -rf ~/.cookiecutters> what is cookies butter cache and how to clear
> if I change my flet version to 0.21.2 will fix this issue?

**Comment by jaychia0418** [comment]
Will try

**Comment by Muddassir-Farooq-official** [comment]
> Will try

Also inform me

**Comment by jaychia0418** [comment]
> > Will try
> 
> Also inform me

I have try cleanup Cookiecutter cache before running flet build command, still same

I upload aab without sign to google play and install, can run properly,
but signed aab still show me white screen

unsign aab can upload to google play internal app sharing only, cannot release to user

build unsign aab command
flet build --project JayCounter --product JayCounter --org jay.chia --include-packages flet_lottie flet_video --build-number 14 --build-version 4.8.2 aab

build signed aab command 
flet build --project JayCounter --product JayCounter --org jay.chia --include-packages flet_lottie flet_video --build-number 14 --build-version 4.8.2 --template-dir C:/Users/chiaj/OneDrive/Desktop/flet-build-template-main/flet-build-template-main aab

unsign aab
https://play.google.com/apps/test/RQOBynJGvow/ahAO29uNTWPR1gSn-_3dBsDFBuMXYC9NNvpXztr76kBpwp6hj2E0FAaeqD6hrcF7obuM3hhbbA8LA2JvWP95A4sIpW 
signed aab
https://play.google.com/apps/test/RQOBynJGvow/ahAO29uNRqk7Pu0xUch-Ya9OGDFG3LP9ERrAwtfvhYQdrAgQ9UMomQCnrkC_a2RH6JguFjEJGE3EEzR3j8ozBdf0Ew

**Comment by Muddassir-Farooq-official** [comment]
What process you apply for generating signed aab

**Comment by jaychia0418** [comment]
I update the android/app/build.gradle to signed aab, add key.properties and chnage buildTypes to release

def keystoreProperties = new Properties()
def keystorePropertiesFile = rootProject.file('key.properties')
if (keystorePropertiesFile.exists()) {
    keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
}

signingConfigs {
        release {
            keyAlias keystoreProperties['keyAlias']
            keyPassword keystoreProperties['keyPassword']
            storeFile keystoreProperties['storeFile'] ? file(keystoreProperties['storeFile']) : null
            storePassword keystoreProperties['storePassword']
        }
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.release
        }
    }
   
like this 
https://github.com/flet-dev/flet/issues/2868

**Comment by jaychia0418** [comment]
I now can install and run properly already by adding this to the build template 
packagingOptions { 
    jniLibs { 
        useLegacyPackaging true
    }
}

But still has one problem, the aab didn't include the package --include-package flet_lottie flet-video
![Screenshot_2024-09-17-12-03-38-454_jay.chia.JayCounter.png](https://github.com/user-attachments/assets/5bbb1b17-1262-43d8-94d3-a390a3de6439)

**Reply by sulasoft** [reply]
Hello, were you able to resolve this?

**Comment by Muddassir-Farooq-official** [comment]
i build aab file but when i install from play store its blank screen

**Comment by jaychia0418** [comment]
> i build aab file but when i install from play store its blank screen

You mean your app also has this problem?

My app now can build an and install from play store, can run properly,
but cannot use video component

**Comment by Muddassir-Farooq-official** [comment]
Can you share your modified build template 
> You mean your app also has this problem?
> 
> My app now can build an and install from play store, can run properly,
> but cannot use video component

**Comment by jaychia0418** [comment]
> Can you share your modified build template

https://drive.google.com/file/d/1dO5enMrLA_TjWv5BGlXs1sqRscOSDq97/view?usp=sharing

**Comment by Muddassir-Farooq-official** [comment]
> > Can you share your modified build template
> 
> https://drive.google.com/file/d/1dO5enMrLA_TjWv5BGlXs1sqRscOSDq97/view?usp=sharing

Can you share me your WhatsApp Number?

**Comment by jaychia0418** [comment]
> > > Can you share your modified build template
> > 
> > 
> > https://drive.google.com/file/d/1dO5enMrLA_TjWv5BGlXs1sqRscOSDq97/view?usp=sharing
> 
> Can you share me your WhatsApp Number?

removed

**Comment by Muddassir-Farooq-official** [comment]
Country code??

**Comment by jaychia0418** [comment]
removed

**Comment by Muddassir-Farooq-official** [comment]
@FeodorFitsner kindly resolve this issue in flet 24.1 of aab file blank screen

**Comment by Muddassir-Farooq-official** [comment]
Any update @FeodorFitsner

**Reply by sulasoft** [reply]
Hi, are you still having issues with a white screen when compiling?

**Reply by Muddassir-Farooq-official** [reply]
Issue in aab file
